c2corg_ui/views/area.py

```diff
@@ -0,0 +1,107 @@
+import logging
+
+from pyramid.renderers import render
+from pyramid.view import view_config
+
+from c2corg_ui.views.document import Document
+
+log = logging.getLogger(__name__)
+
+
+class Area(Document):
+
+    _API_ROUTE = 'areas'
+
+    @view_config(route_name='areas_index')
+    def index(self):
+        return self._index('c2corg_ui:templates/area/index.html')
+
+    @view_config(route_name='areas_sitemap',
+                 renderer='c2corg_ui:templates/area/sitemap.html')
+    @view_config(route_name='areas_sitemap_default',
+                 renderer='c2corg_ui:templates/area/sitemap.html')
+    def sitemap(self):
+        areas, total, filter_params, lang = self._get_documents()
+        self.template_input.update({
+            'areas': areas,
+            'total': total,
+            'filter_params': filter_params,
+            'lang': lang
+        })
+        return self.template_input
+
+    @view_config(route_name='areas_view_id')
+    @view_config(route_name='areas_view_id_lang')
+    def redirect_to_full_url(self):
+        self._redirect_to_full_url()
+
+    @view_config(route_name='areas_view')
+    def detail(self):
+        id, lang = self._validate_id_lang()
+
+        def render_page(area, locale):
+            # FIXME self._get_geometry(area['geometry']['geom_detail'])
+            # always returns None!?
```

Comment:
But it is returned from the API? http://c2corgv6-api.demo-camptocamp.com/areas/523281